internal/pkg/action/operator_uninstall.go

```diff
@@ -74,41 +66,68 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 		return fmt.Errorf("get subscription CSV %q: %v", csvName, err)
 	}
 
+	// Subscriptions can be deleted asynchronously.
+	if err := u.deleteObjects(ctx, sub); err != nil {
+		return err
+	}
+
+	// If we could not find a csv associated with the subscription, that likely
+	// means there is no CSV associated with it yet. Return a nil error.
+	if csv == nil {
+		u.Logf("csv for package %q not found", u.Package)
+		return nil
+	}
```

Comment:
Returning early here means we don't get a chance to see if we can delete operator groups or other bundle artifacts that may have been deployed separately from the CSV (e.g. eventually CRDs, hopefully). We will likely change operator group handling in a follow-on PR, but in general, I think we should gracefully handle missing CSVs and just skip the steps that require them.

As an example, I would expect the following output if an install failed to deploy an operator because of a non-existent version. In this case a CSV never gets created, but uninstall figures out what's present and deletes what exists:
```console
$ kubectl operator install etcd -v 1.2.3 -w etcd --create-operator-group
operatorgroup "etcd" created
subscription "etcd" created
failed to install operator: waiting for install plan to exist: timed out waiting for the condition

$ kubectl get operatorgroup,subscription
NAME                                      AGE
operatorgroup.operators.coreos.com/etcd   86s

NAME                                     PACKAGE   SOURCE                  CHANNEL
subscription.operators.coreos.com/etcd   etcd      operatorhubio-catalog

$ kubectl operator uninstall etcd --delete-operator-groups
subscription "etcd" deleted
operatorgroup "etcd" deleted
operator "etcd" uninstalled
```

Reply:
For the list-operands problem, we should talk more about what should happen when a CSV is not present. It is possible to have operands without the CSV. But it is currently not possible to know what operands are associated with an operator without a CSV.

Reply:
Ok, thanks for that example, I think I'm following now. I will look to add this in